types/ioTracker/ioTracker3/uplink.js

```diff
@@ -0,0 +1,512 @@
+            break;
+
+          case 0x02:
+            beacon = parseBluetoothBeacons02();
+            break;
+
+          default:
+            return { errors: ["Invalid addSlotInfo type"] };
+        }
+        if (beacon === null) {
+          return { errors: ["Invalid beacon type"] };
+        }
+
+        decoded.bluetoothInfo.beacons.push(beacon);
+      }
+    }
+
+    if (decoded.sensorContent.containsRelativeHumidity) {
+      decoded.relativeHumidity =
+        toUnsignedShort(bytes[index++], bytes[index++]) / 100;
+    }
+
+    if (decoded.sensorContent.containsAirPressure) {
+      decoded.airPressure =
+        (bytes[index++] << 16) + (bytes[index++] << 8) + bytes[index++];
+    }
+
+    if (decoded.sensorContent.containsManDown) {
+      const manDownData = bytes[index++];
+      const manDownState = manDownData & 0x0f;
+      let manDownStateLabel;
+      switch (manDownState) {
+        case 0x00:
+          manDownStateLabel = "ok";
+          break;
+        case 0x01:
+          manDownStateLabel = "sleeping";
+          break;
+        case 0x02:
+          manDownStateLabel = "preAlarm";
+          break;
+        case 0x03:
+          manDownStateLabel = "alarm";
+          break;
+        default:
+          manDownStateLabel = `${manDownState}`;
+          break;
+      }
+      decoded.manDown = {
+        state: manDownStateLabel,
+        positionAlarm: !!(manDownData & 0x10),
+        movementAlarm: !!(manDownData & 0x20),
+      };
+    }
+  }
+
+  if (decoded.containsGps) {
+    decoded.gps = {};
+    decoded.gps.navStat = bytes[index++];
+    decoded.gps.latitude =
+      toSignedInteger(
+        bytes[index++],
+        bytes[index++],
+        bytes[index++],
+        bytes[index++],
+      ) / 10000000;
+    decoded.gps.longitude =
+      toSignedInteger(
+        bytes[index++],
+        bytes[index++],
+        bytes[index++],
+        bytes[index++],
+      ) / 10000000;
+    decoded.gps.altRef = toUnsignedShort(bytes[index++], bytes[index++]) / 10;
+    decoded.gps.hAcc = bytes[index++];
+    decoded.gps.vAcc = bytes[index++];
+    decoded.gps.sog = toUnsignedShort(bytes[index++], bytes[index++]) / 10;
+    decoded.gps.cog = toUnsignedShort(bytes[index++], bytes[index++]) / 10;
+    decoded.gps.hdop = bytes[index++] / 10;
+    decoded.gps.numSvs = bytes[index++];
+  }
+
+  return decoded;
+}
+
+function consume(event) {
+  const payload = event.data.payloadHex;
+  const decoded = decoder(Hex.hexToBytes(payload));
+
+  if (decoded.gps !== undefined) {
+    if (Object.keys(decoded.gps).length > 0) {
+      emit("sample", { data: decoded.gps, topic: "gps" });
+      delete decoded.gps;
+    }
+  }
+
+  // Redundant data is only used as flags
+  delete decoded.containsGps;
+  delete decoded.containsOnboardSensors;
+  delete decoded.containsSpecial;
+
+  if (decoded.sensorContent !== undefined) {
+    delete decoded.sensorContent;
+  }
+
+  emit("sample", { data: decoded, topic: "default" });
```

Comment:
it is clearer if you build the sample by specifying keys and values instead of just dropping keys